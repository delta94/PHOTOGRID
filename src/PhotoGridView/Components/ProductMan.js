import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { AssignedPhoto } from '../Partials/AssignedPhoto';
import { AddProductsBtn } from '../Partials/AddProductsBtn';
import { AddProductsModal} from '../Partials/AddProductsModal';

export class ProductMan extends Component {
  constructor() {
    super();

    this.state = {
      modalVisible: false
    };

    this.handleClickCloseBtn = this.handleClickCloseBtn.bind(this);
    this.handleToggleModal = this.handleToggleModal.bind(this);
  }

  handleClickCloseBtn(id) {
    const products = this.props.products;
    const index = products.indexOf(id);
    products.splice(index, 1);
    this.props.onProductsChange(products);
  }

  handleToggleModal() {
    this.setState({
      modalVisible: !this.state.modalVisible
    });
  }

  render() {
    const { 
      allProducts, 
      products
    } = this.props;
    const { modalVisible } = this.state;
    const length = products.length;

    return (
      <div 
        className="photo-grid-productman-container"
      > 
        <div
          className={`d-flex align-items-center ${length > 3 && 'flex-grow-1'}`}
        >
          {
            products && products.map((id, index) => {
              const product = allProducts.find((item) => item.id === id);
              return (
                index < 3 &&
                <AssignedPhoto
                  id = {product.id}
                  title = {product.title}
                  image = {product.image}
                  key = {id}
                  handleClickCloseBtn = {this.handleClickCloseBtn}
                />
              );
            })
          }
          {
            length > 3 && (
              <span
                className="color__grey__light font-size_tiny ml-1"
              >
                {length - 3} more
              </span>
            )
          }       
        </div>
        <div
          className="d-flex align-items-center ml-1 position-relative"
          style={{ height: '80%' }}
        >
          <AddProductsBtn
            handleToggleModal={this.handleToggleModal}
          />
          {
            modalVisible &&
            <AddProductsModal
              allProducts={allProducts}
              products={products}
              handleToggleModal={this.handleToggleModal}
              onProductsChange={this.props.onProductsChange}
            />
          }
        </div>
        
      </div>
    );
  }
}

ProductMan.propTypes = {
  allProducts: PropTypes.array,
  products: PropTypes.array,
  onProductsChange: PropTypes.func
};

export default ProductMan;
