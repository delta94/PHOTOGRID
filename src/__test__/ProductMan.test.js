import React from 'react';
import { shallow, mount } from 'enzyme';

import ProductMan from '../PhotoGridView/Components/ProductMan';

const mockData = {
  photo: {
    id: 1,
    imageUrl: 'https://scontent-ber1-1.cdninstagram.com/vp/88f4d9772eff7bc7e429e3629f507339/5C8BE025/t51.2885-15/e35/44887346_2211003352244412_5881337131845119437_n.jpg',
    products: [1],
  },
  products: [
    {
      id: 1,
      title: 'Whisker',
      image: 'https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg'
    },
    {
      id: 2,
      title: 'Merry',
      image: 'https://ae01.alicdn.com/kf/HTB1Cvrtl8jTBKNjSZFuq6z0HFXaa/Automatic-Toothpas…oothbrush-Holder-with-Cups-No-Nail-Wall-Stand-Shelf-Bathroom-Organizer.jpg'
    },
    {
      id: 3,
      title: 'Apple',
      image: 'https://img.staticbg.com/thumb/large/oaupload/banggood/images/9B/D2/54eafdb3-7f9c-4d56-9522-73c12c2fe2c3.jpg'
    },
    {
      id: 4,
      title: 'Sample',
      image: 'https://ae01.alicdn.com/kf/HTB1KwN_cGmgSKJjSsplq6yICpXai/Gamiss-Spring-Autu…ual-Tshirt-Plus-Size-Halloween-Batwing-Loose-Long-Sleeve-Solid-Color-T.jpg'
    }
  ]
};
const mockhandler = jest.fn();

describe('ProductMan Component', () => {
  const wrapper = mount(
    <ProductMan
      products={mockData.photo.products}
      allProducts={mockData.products}
      onProductsChange={mockhandler}
    />
  );

  const addProductButton = wrapper.find('AddProductsBtn');
  it('allows us to set props for ProductMan', () => {
    expect(wrapper.props().allProducts).toBe(mockData.products);
    expect(wrapper.props().products).toBe(mockData.photo.products);
  });

  it('click add product button', () => {
    expect(wrapper.find('AddProductsModal')).toHaveLength(0);
    addProductButton.find('div').simulate('click');
    expect(wrapper.find('AddProductsModal')).toHaveLength(1);
  });

});

describe('ProductMan SnapShot', () => {
  const wrapper = shallow(
    <ProductMan
      products={mockData.photo.products}
      allProducts={mockData.products}
      onProductsChange={mockhandler}
    />
  );

  it('match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});