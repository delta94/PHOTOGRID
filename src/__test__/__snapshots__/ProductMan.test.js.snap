// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductMan SnapShot match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductMan
    allProducts={
      Array [
        Object {
          "id": 1,
          "image": "https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg",
          "title": "Whisker",
        },
        Object {
          "id": 2,
          "image": "https://ae01.alicdn.com/kf/HTB1Cvrtl8jTBKNjSZFuq6z0HFXaa/Automatic-Toothpas…oothbrush-Holder-with-Cups-No-Nail-Wall-Stand-Shelf-Bathroom-Organizer.jpg",
          "title": "Merry",
        },
        Object {
          "id": 3,
          "image": "https://img.staticbg.com/thumb/large/oaupload/banggood/images/9B/D2/54eafdb3-7f9c-4d56-9522-73c12c2fe2c3.jpg",
          "title": "Apple",
        },
        Object {
          "id": 4,
          "image": "https://ae01.alicdn.com/kf/HTB1KwN_cGmgSKJjSsplq6yICpXai/Gamiss-Spring-Autu…ual-Tshirt-Plus-Size-Halloween-Batwing-Loose-Long-Sleeve-Solid-Color-T.jpg",
          "title": "Sample",
        },
      ]
    }
    onProductsChange={[MockFunction]}
    products={
      Array [
        1,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="d-flex align-items-center false"
        >
          <AssignedPhoto
            handleClickCloseBtn={[Function]}
            id={1}
            image="https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg"
            title="Whisker"
          />
        </div>,
        <div
          className="d-flex align-items-center ml-1 position-relative"
          style={
            Object {
              "height": "80%",
            }
          }
        >
          <AddProductsBtn
            handleToggleModal={[Function]}
          />
        </div>,
      ],
      "className": "photo-grid-productman-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <AssignedPhoto
                handleClickCloseBtn={[Function]}
                id={1}
                image="https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg"
                title="Whisker"
              />,
            ],
            false,
          ],
          "className": "d-flex align-items-center false",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "handleClickCloseBtn": [Function],
              "id": 1,
              "image": "https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg",
              "title": "Whisker",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <AddProductsBtn
              handleToggleModal={[Function]}
            />,
            false,
          ],
          "className": "d-flex align-items-center ml-1 position-relative",
          "style": Object {
            "height": "80%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleToggleModal": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="d-flex align-items-center false"
          >
            <AssignedPhoto
              handleClickCloseBtn={[Function]}
              id={1}
              image="https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg"
              title="Whisker"
            />
          </div>,
          <div
            className="d-flex align-items-center ml-1 position-relative"
            style={
              Object {
                "height": "80%",
              }
            }
          >
            <AddProductsBtn
              handleToggleModal={[Function]}
            />
          </div>,
        ],
        "className": "photo-grid-productman-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <AssignedPhoto
                  handleClickCloseBtn={[Function]}
                  id={1}
                  image="https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg"
                  title="Whisker"
                />,
              ],
              false,
            ],
            "className": "d-flex align-items-center false",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "handleClickCloseBtn": [Function],
                "id": 1,
                "image": "https://ae01.alicdn.com/kf/HTB1N8Q9G7yWBuNjy0Fpq6yssXXal/Led-twig-lights-holiday-lights-branches-Christmas-ornaments-HJW.jpg",
                "title": "Whisker",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <AddProductsBtn
                handleToggleModal={[Function]}
              />,
              false,
            ],
            "className": "d-flex align-items-center ml-1 position-relative",
            "style": Object {
              "height": "80%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleToggleModal": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
